task_processing/plugins/mesos/retrying_executor.py

```diff
@@ -87,12 +96,13 @@ def retry_loop(self):
 
             time.sleep(1)
 
-    def run(self, task_config):
-        if task_config.task_id not in self.task_retries:
+    def run(self, task_config, task_id):
+        if task_id not in self.task_retries:
             with self.task_retries_lock:
                 self.task_retries = self.task_retries.set(
-                    task_config.task_id, self.retries)
-        self.executor.run(task_config)
+                    task_id, self.retries)
+
+        self.executor.run(task_config, self.grow(task_id))
```

Comment:
What if a user calls run() with the same task_id? It results in one retry without task_retries map being updated. More importantly, locking is needed to avoid races with the retry loop.